avm/res/devops-infrastructure/managed-devops-pools/main.bicep

```diff
@@ -0,0 +1,448 @@
+type organizationProfileType = organizationGitHubType | organizationAzureDevOpsType
+
+type secretManagementType = {
+  @description('Required. The secret management settings of the machines in the pool.')
+  keyExportable: bool
+
+  @description('Required. The list of certificates to install on all machines in the pool.')
+  observedCertificates: string[]
+
+  @description('Optional. Where to store certificates on the machine.')
+  certificateStoreLocation: string?
+}?
+
+type dataDiskType = {
+  @description('Optional. The type of caching to be enabled for the data disks. The default value for caching is readwrite. For information about the caching options see: https://blogs.msdn.microsoft.com/windowsazurestorage/2012/06/27/exploring-windows-azure-drives-disks-and-images/.')
+  caching: ('None' | 'ReadOnly' | 'ReadWrite')?
+
+  @description('Optional. The initial disk size in gigabytes.')
+  diskSizeGiB: int?
+
+  @description('Optional. The drive letter for the empty data disk. If not specified, it will be the first available letter. Letters A, C, D, and E are not allowed.')
+  driveLetter: string?
+
+  @description('Optional. The storage Account type to be used for the data disk. If omitted, the default is Standard_LRS.')
+  storageAccountType: ('Premium_LRS' | 'Premium_ZRS' | 'StandardSSD_LRS' | 'StandardSSD_ZRS' | 'Standard_LRS')?
+}[]?
+
+type resourcePredictionsProfileAutomaticType = {
+  @description('Required. The stand-by agent scheme is determined based on historical demand.')
+  kind: 'Automatic'
+
+  @description('Required. Determines the balance between cost and performance.')
+  predictionPreference: 'Balanced' | 'MostCostEffective' | 'MoreCostEffective' | 'MorePerformance' | 'BestPerformance'
+}
+
+type resourcePredictionsProfileManualType = {
+  @description('Required. Customer provides the stand-by agent scheme.')
+  kind: 'Manual'
+}
+
+type agentStatefulType = {
+  @description('Required. Stateful profile meaning that the machines will be returned to the pool after running a job.')
+  kind: 'Stateful'
+
+  @description('Required. How long should stateful machines be kept around. The maximum is one week.')
+  maxAgentLifetime: string
+
+  @description('Required. How long should the machine be kept around after it ran a workload when there are no stand-by agents. The maximum is one week.')
+  gracePeriodTimeSpan: string
+
+  @description('Optional. Defines pool buffer/stand-by agents.')
+  resourcePredictions: object?
+
+  @discriminator('kind')
+  @description('Optional. Determines how the stand-by scheme should be provided.')
+  resourcePredictionsProfile: (resourcePredictionsProfileAutomaticType | resourcePredictionsProfileManualType)?
+}
+
+type agentStatelessType = {
+  @description('Required. Stateless profile meaning that the machines will be cleaned up after running a job.')
+  kind: 'Stateless'
+
+  @description('Optional. Defines pool buffer/stand-by agents.')
+  resourcePredictions: {
+    timeZone: (
```

Comment:
@AlexanderSehr I want to know about your opinion surrounding the timeZone property. The current list is just a subset of the actual timeZone list (output from https://learn.microsoft.com/en-us/dotnet/api/system.timezoneinfo.getsystemtimezones?view=net-8.0).

I planned to add all timeZones to the UDT, so that the user can select the preferred timezone instead of needing to look it up. It will be an extensive list though. Would you go this way too?

Reply:
Hey @johnlokerse,
a good question. If the below list is correct, we're looking at 141 entries:

<details>
<summary>TZs</summary>

```
> (Get-TimeZone -ListAvailable).StandardName
Dateline Standard Time
UTC-11
Aleutian Standard Time
Hawaiian Standard Time
Marquesas Standard Time
Alaskan Standard Time
UTC-09
Pacific Standard Time (Mexico)
UTC-08
Pacific Standard Time
US Mountain Standard Time
Mountain Standard Time (Mexico)
Mountain Standard Time
Yukon Standard Time
Central America Standard Time
Central Standard Time
Easter Island Standard Time
Central Standard Time (Mexico)
Canada Central Standard Time
SA Pacific Standard Time
Eastern Standard Time (Mexico)
Eastern Standard Time
Haiti Standard Time
Cuba Standard Time
US Eastern Standard Time
Turks and Caicos Standard Time
Paraguay Standard Time
Atlantic Standard Time
Venezuela Standard Time
Central Brazilian Standard Time
SA Western Standard Time
Pacific SA Standard Time
Newfoundland Standard Time
Tocantins Standard Time
E. South America Standard Time
SA Eastern Standard Time
Argentina Standard Time
Montevideo Standard Time
Magallanes Standard Time
Saint Pierre Standard Time
Bahia Standard Time
UTC-02
Greenland Standard Time
Mid-Atlantic Standard Time
Azores Standard Time
Cabo Verde Standard Time
Coordinated Universal Time
GMT Standard Time
Greenwich Standard Time
Sao Tome Standard Time
Morocco Standard Time
W. Europe Standard Time
Central Europe Standard Time
Romance Standard Time
Central European Standard Time
W. Central Africa Standard Time
GTB Standard Time
Middle East Standard Time
Egypt Standard Time
E. Europe Standard Time
West Bank Gaza Standard Time
South Africa Standard Time
FLE Standard Time
Jerusalem Standard Time
South Sudan Standard Time
Russia TZ 1 Standard Time
Sudan Standard Time
Libya Standard Time
Namibia Standard Time
Jordan Standard Time
Arabic Standard Time
Syria Standard Time
Turkey Standard Time
Arab Standard Time
Belarus Standard Time
Russia TZ 2 Standard Time
E. Africa Standard Time
Volgograd Standard Time
Iran Standard Time
Arabian Standard Time
Astrakhan Standard Time
Azerbaijan Standard Time
Russia TZ 3 Standard Time
Mauritius Standard Time
Saratov Standard Time
Georgian Standard Time
Caucasus Standard Time
Afghanistan Standard Time
West Asia Standard Time
Qyzylorda Standard Time
Russia TZ 4 Standard Time
Pakistan Standard Time
India Standard Time
Sri Lanka Standard Time
Nepal Standard Time
Central Asia Standard Time
Bangladesh Standard Time
Omsk Standard Time
Myanmar Standard Time
SE Asia Standard Time
Altai Standard Time
W. Mongolia Standard Time
Russia TZ 6 Standard Time
Novosibirsk Standard Time
Tomsk Standard Time
China Standard Time
Russia TZ 7 Standard Time
Malay Peninsula Standard Time
W. Australia Standard Time
Taipei Standard Time
Ulaanbaatar Standard Time
Aus Central W. Standard Time
Transbaikal Standard Time
Tokyo Standard Time
North Korea Standard Time
Korea Standard Time
Russia TZ 8 Standard Time
Cen. Australia Standard Time
AUS Central Standard Time
E. Australia Standard Time
AUS Eastern Standard Time
West Pacific Standard Time
Tasmania Standard Time
Russia TZ 9 Standard Time
Lord Howe Standard Time
Bougainville Standard Time
Russia TZ 10 Standard Time
Magadan Standard Time
Norfolk Standard Time
Sakhalin Standard Time
Central Pacific Standard Time
Russia TZ 11 Standard Time
New Zealand Standard Time
UTC+12
Fiji Standard Time
Kamchatka Standard Time
Chatham Islands Standard Time
UTC+13
Tonga Standard Time
Samoa Standard Time
Line Islands Standard Time
```

</details> 

Aside the fundemental question whether it's wasting previous kilobytes, I'm actually wondering if it's a usable list given its size. To validate this, I added the parameter `timeZone` to a random bicep file to then reference set parameter:
```bicep
@allowed([
  'Dateline Standard Time'
  'UTC-11'
  'Aleutian Standard Time'
  'Hawaiian Standard Time'
  'Marquesas Standard Time'
  'Alaskan Standard Time'
  'UTC-09'
  'Pacific Standard Time (Mexico)'
  'UTC-08'
  (...)
])
param timezone string
```
![image](https://github.com/user-attachments/assets/59bdff9f-18e8-4fd3-9e9b-3120770dd01a)

The good news - you can search in the field. The bad - even with it, assuming I don't know exactly what I looking for / it has a different name, it took me quite a few retries. For example 'BST / British Standard Time' is not listed.

Truth be told, I'm more leaning towards **not** including the allowed set, but instead mention in the description either a link to docs that list all TZs that can be set, or as above, provide the pwsh command to get the same: `(Get-TimeZone -ListAvailable).Id`.

Reply:
I agree with not including it. I also see duplicated such as `AUS Central Standard Time` and `Cen. Australia Standard Time`... better to keep the user informed via the description.